File: 1884-minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.java
class Solution {
    public int minOperations(String s) {
        int result1=0,result2=0;
        for(int i=0;i<s.length();i++)
        {
            if(i%2==0)
            {
                if(s.charAt(i)=='0') result1+=1;
                else result2+=1;
            }

            if(i%2==1)
            {
                if(s.charAt(i)=='1')result1+=1;
                else result2+=1;

            }
        }
        return Math.min(result1,result2);
    }
}

File: 1884-minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
    let result1=0, result2 = 0;
    let alter=true;
    let arr=s.split("");
    let arr2=s.split("");
    for(let i=0;i<arr.length;i++){
        if(arr[i] == alter){
            result1++;
            arr[i]=Number(!alter);
        } 
        if(arr2[i] == (!alter)){
            result2++;
            arr[i]=Number(alter);
        } 
        alter=!alter;
    }
    return Math.min(result1,result2);
};